src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskCache.cs

```diff
@@ -62,4 +41,11 @@ private static Task<int>[] CreateInt32Tasks()
             return tasks;
         }
     }
+
+    /// <summary>Provides a cache of tasks for async methods.</summary>
+    internal static class TaskCache<TResult>
+    {
+        /// <summary>A cached task for default(TResult).</summary>
+        internal static readonly Task<TResult> s_defaultResultTask = TaskCache.CreateCacheableTask<TResult>(default);
```

Comment:
It may be better for footprint of typical apps to attach this static to an existing type like it was done before, even if it may not be used.

Type instantiation costs 100s of bytes in footprint. Reference types (where this cache kicks in) are used more frequently than value types.

Reply:
Ok.  I can move it to `Task<TResult>` if you think that's better.